### PR TITLE
feat: Add SchedulerTask which will be notified once cancelled.

### PR DESCRIPTION
**Pull request by He-Pin**
Motivation:
refs: https://github.com/apache/pekko/issues/1579

Tests is still missing

### CODE REVIEW

File: actor/src/main/scala/org/apache/pekko/actor/LightArrayRevolverScheduler.scala [hunk 1]
```diff
@@ -390,7 +395,19 @@ object LightArrayRevolverScheduler {
...
+        true
+      case _ => true
+    }
+
+    private def notifyCancellation(task: SchedulerTask): Unit = {
+      try {
+        task.cancelled()
+      } catch {
+        case Throwable =>
+        // TODO how to logging here?
```
Comment (He-Pin) [inline actor/src/main/scala/org/apache/pekko/actor/LightArrayRevolverScheduler.scala:409]: Can't access the `loggingAdapter` here :(

Comment (He-Pin) [reply actor/src/main/scala/org/apache/pekko/actor/LightArrayRevolverScheduler.scala:409]: I think these method should run on a dedicated executor?

**Comment by pjfanning** [comment]
I'm not sure that we could add this in a v1.1 patch. I'd prefer if we delay this till v1.2.0.

**Comment by He-Pin** [comment]
I'm fine with this, but as the current method is add as a default method, it should be fine too.

**Review by pjfanning — APPROVED** [review]
lgtm

File: actor-tests/src/test/scala/org/apache/pekko/actor/SchedulerSpec.scala [hunk 2]
```diff
@@ -131,6 +131,17 @@ trait SchedulerSpec extends BeforeAndAfterEach with DefaultTimeout with Implicit
         task.isCancelled should ===(true)
       }
 
+      "notify callback if cancel is performed before execution" taggedAs TimingTest in {
+        val latch = new CountDownLatch(1)
+        val task = system.scheduler.scheduleOnce(10 seconds,
```
Comment (pjfanning) [inline actor-tests/src/test/scala/org/apache/pekko/actor/SchedulerSpec.scala:136]: can we have a shorter schedule? to save time on the test run

**Comment by pjfanning** [comment]
I'm quite worried that these tests will be brittle in CI. The CI agents can often suffer from performance issues and tests can take longer to run than on laptops, etc.

File: actor/src/main/scala/org/apache/pekko/actor/Scheduler.scala [hunk 3]
```diff
@@ -498,6 +503,20 @@ trait Scheduler {
 // this one is just here so we can present a nice AbstractScheduler for Java
 abstract class AbstractSchedulerBase extends Scheduler
 
+/**
+ * A Task that will be notified when it is cancelled.
+ *
+ * @since 1.2.0
```
Comment (He-Pin) [inline actor/src/main/scala/org/apache/pekko/actor/Scheduler.scala:509]: add the since annotation

**Pull request merged**